src/main/java/parsing/Parser.java

```diff
@@ -379,4 +411,31 @@ private boolean determineBasis(String line, String[] allGenomes) {
         }
         return result;
     }
+
+    /**
+     * Reads the genome paths from the cache and puts these in an accessible array.
+     * @param path the path of the genome cache file.
+     */
+    private void readGenomes(String path) {
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(path));
+            int nol = getNumberOfLine(new File(path));
+            String line = br.readLine();
+            String[] genomes = line.split("\t");
+            GraphInfo.getInstance().setGenomesNum(Integer.parseInt(genomes[0]));
+            int[][] paths = new int[nol - 2][];
+            for (int i = 0; i < nol - 2; i++) {
+                line = br.readLine();
+                genomes = line.split("\t");
+                int[] genPath = new int[Integer.parseInt(genomes[0])];
+                for (int j = 0; j < Integer.parseInt(genomes[0]); j++) {
+                    genPath[j] = Integer.parseInt(genomes[j + 1]);
+                }
+                paths[i] = genPath;
```

Comment:
Naming is quite ambiguous now that you mention it. I'll think of a better name for the array.